sdk/java/src/main/java/com/chain/api/CoreConfig.java

```diff
@@ -0,0 +1,141 @@
+package com.chain.api;
+
+import com.chain.http.*;
+import com.chain.exception.*;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.*;
+
+public class CoreConfig {
+
+  public static class Info {
+    public String state;
+
+    @SerializedName("is_configured")
+    public Boolean isConfigured;
+
+    @SerializedName("configured_at")
+    public Date configuredAt;
+
+    @SerializedName("is_signer")
+    public Boolean isSigner;
+
+    @SerializedName("is_generator")
+    public Boolean isGenerator;
+
+    @SerializedName("generator_url")
+    public String generatorUrl;
+
+    @SerializedName("generator_access_token")
+    public String generatorAccessToken;
+
+    @SerializedName("blockchain_id")
+    public String blockchainId;
+
+    @SerializedName("block_height")
+    public long blockHeight;
+
+    @SerializedName("generator_block_height")
+    public long generatorBlockHeight;
+
+    @SerializedName("generator_block_height_fetched_at")
+    public Date generatorBlockHeightFetchedAt;
+
+    @SerializedName("network_rpc_version")
+    public int networkRpcVersion;
+
+    @SerializedName("core_id")
+    public String coreId;
+
+    public String version;
+
+    @SerializedName("build_commit")
+    public String buildCommit;
+
+    @SerializedName("build_date")
+    public String buildDate;
+
+    @SerializedName("build_config")
+    public BuildConfig buildConfig;
+
+    public Map<String, Object> health;
+
+    public Snapshot snapshot;
+
+    public static class BuildConfig {
+      @SerializedName("is_loopback_auth")
+      public Boolean isLoopbackAuth;
+
+      @SerializedName("is_mockhsm")
+      public Boolean isMockHsm;
+
+      @SerializedName("is_reset")
+      public Boolean isReset;
+    }
+
+    public static class Snapshot {
+      public int attempt;
+
+      public long height;
+
+      public long size;
+
+      public long downloaded;
+
+      @SerializedName("in_progress")
+      public Boolean inProgress;
+    }
+  }
+
+  public static Info getInfo(Client client) throws ChainException {
+    return client.request("info", null, Info.class);
+  }
+
+  public static void reset(Client client) throws ChainException {
+    client.request("reset", null, SuccessMessage.class);
+  }
+
+  public static void resetEverything(Client client) throws ChainException {
+    Map<String, Boolean> params = new HashMap<>();
+    params.put("everything", true);
+    client.request("reset", params, SuccessMessage.class);
+  }
+
+  public static class Builder {
+    @SerializedName("is_generator")
+    private Boolean isGenerator;
```

Comment:
Hmm, very interesting! TBH I typed `Boolean` with the capital B more or less by accident, but now that you have me thinking, I suspect that capital-B `Boolean` may in fact be a bit better in this case.

> I'm assuming we actually want this to default to false

I suspect this is not accurate. There's no particular reason `false` is better than `true` when there is no value  returned from the server. Defaults are okay as a convenience mechanism for input, but they seem irrelevant when we're talking about output.

What `null` would imply here is "undefined", "no information", or even "there's a bug in the response-generating code and you should not rely on this value". I would actually love to be able to mark some (but not all) of these fields as "required", so that the JSON deserializer would throw an exception if the serialized version doesn't contain certain fields. Unfortunately, I dont think [GSON will be adding that any time soon](https://github.com/google/gson/issues/61).

FWIW, what will happen here if you try to use a `null` Boolean object is a null pointer exception, which I think is ultimately better than arbitrarily assuming falsiness. Admittedly it's the lesser of two evils.